pkg/git/gogit/transport.go

```diff
@@ -88,16 +89,24 @@ func (s *PublicKeyAuth) Method(secret corev1.Secret) (*git.Auth, error) {
 		user = git.DefaultPublicKeyAuthUser
 	}
 
-	password := secret.Data["password"]
-	pk, err := ssh.NewPublicKeys(user, identity, string(password))
+	password, ok := secret.Data["password"]
+	var signer cryptossh.Signer
+	var err error
+	if ok {
+		signer, err = cryptossh.ParsePrivateKeyWithPassphrase(identity, password)
+	} else {
+		signer, err = cryptossh.ParsePrivateKey(identity)
+	}
```

Comment:
```suggestion
    signer, err := ssh.ParsePrivateKey(identity)
	if _, ok := err.(*ssh.PassphraseMissingError); ok {
		signer, err = ssh.ParsePrivateKeyWithPassphrase(identity, secret.Data["password"])
	}
	if err != nil {
		return nil, err
	}
```

may be a bit cleaner, but no strong opinion.